### PR TITLE
[REC-740] Get Mandate Constraints + tests

### PR DESCRIPTION
**Pull request by tl-andrei-sorbun**
# Description

This PR adds the GetConstraints call as documented [here](https://docs.truelayer.com/reference/get-constraints) to the SDK. 

## Type of change

Please select multiple options if required.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the relevant documentation

### CODE REVIEW

File: src/main/java/com/truelayer/java/mandates/entities/GetConstraintsResponse.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+@Builder
+@Getter
+@ToString
+@EqualsAndHashCode
```
Comment (dili91) [inline src/main/java/com/truelayer/java/mandates/entities/GetConstraintsResponse.java:11]: Unless we build this somewhere (usually we don't for responses) you can simply use `@Value`, which includes all these (except the Builder)

File: src/test/java/com/truelayer/java/acceptance/MandatesAcceptanceTests.java [hunk 2]
```diff
@@ -155,6 +159,121 @@ public void itShouldGetFunds() {
...
+        assertNotError(createMandateResponse);
+
+        // save the params to be used later
+        Integer paymentAmount = 1;
+        Integer calculatedRemainingPerMonth = createMandateRequest
+                        .getConstraints()
+                        .getPeriodicLimits()
+                        .getMonth()
+                        .getMaximumAmount()
+                - paymentAmount;
```
Comment (dili91) [inline src/test/java/com/truelayer/java/acceptance/MandatesAcceptanceTests.java:182]: I would not bother doing calculations here. from the perspective of the library the test is passed if responses are aligned with the API refs.

File: src/main/java/com/truelayer/java/mandates/entities/GetConstraintsResponse.java (hunk 1 above)
Comment (tl-andrei-sorbun) [reply src/main/java/com/truelayer/java/mandates/entities/GetConstraintsResponse.java:11]: I'm using the builder in the acceptance test 😅

**Review by dili91 — APPROVED** [review]
Thanks for taking care ⭐

**Pull request merged**